x-pack/platform/plugins/shared/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -81,22 +81,33 @@ const statsRoute = createDatasetQualityServerRoute({
       return dataStream.userPrivileges.canMonitor;
     });
 
-    const dataStreamsStats = isServerless
-      ? await getDataStreamsMeteringStats({
-          esClient: esClientAsSecondaryAuthUser,
-          dataStreams: privilegedDataStreams.map((stream) => stream.name),
-        })
-      : await getDataStreamsStats({
-          esClient,
-          dataStreams: privilegedDataStreams.map((stream) => stream.name),
-        });
+    const dataStreamsNames = privilegedDataStreams.map((stream) => stream.name);
+    const [dataStreamsStats, dataStreamsCreationDate] = await Promise.all([
+      isServerless
+        ? getDataStreamsMeteringStats({
+            esClient: esClientAsSecondaryAuthUser,
+            dataStreams: dataStreamsNames,
+          })
+        : getDataStreamsStats({
+            esClient,
+            dataStreams: dataStreamsNames,
+          }),
+
+      params.query.includeCreationDate
```

Comment:
Could you add some tests for this (`x-pack/test/dataset_quality_api_integration/tests/data_streams/stats.spec.ts`)?

Reply:
added in https://github.com/elastic/kibana/pull/210301/commits/39796925bff9f5d0ef388922426fbaab5d17adc4